backend/papersfeed/utils/recommendations/utils.py

```diff
@@ -83,14 +85,59 @@ def select_recommendation(args):
 
     recommendation_queryset = UserRecommendation.objects.filter(user_id=request_user.id)
 
-    recommendations = get_results_from_queryset(recommendation_queryset, 10, page_number)
+    recommendations = get_results_from_queryset(recommendation_queryset, 20, page_number)
 
     is_finished = not recommendations.has_next()
 
     recommendations = __pack_recommendations(recommendations, request_user)
 
     return recommendations, page_number, is_finished
 
+def select_keyword_init(args):
+    """Get keywords init"""
+
+    page_number = 1 if constants.PAGE_NUMBER not in args else int(args[constants.PAGE_NUMBER])
+
+    keyword_queryset = (Keyword.objects.annotate(num_papers=Count('papers'))
+                        .filter(num_papers__gt=9, num_papers__lt=100))
+
+    keywords = get_results_from_queryset(keyword_queryset, 20, page_number)
+
+    is_finished = not keywords.has_next()
+
+    keywords = paper_utils.pack_keywords(keywords)
+
+    return keywords, page_number, is_finished
+
+def insert_recommendation_init(args):
+    """Get recommendation init"""
+
+    request_user = args[constants.USER]
+    keywords = args[constants.KEYWORDS]
+
+    for k, keyword in enumerate(keywords):
+        paper_queryset = Keyword.objects.filter(Q(id=keyword))[0].papers.all()
+        paper_ids = [paper.id for paper in paper_queryset]
+
+        paper_counts = paper_utils.get_paper_like_count(paper_ids, 'paper_id')
+
+        if paper_counts:
+            paper_sort = sorted(paper_counts.items(), key=(lambda x: x[1]), reverse=True)
```

Comment:
I think there may be more Django or MySQL-tic way to do it. We can deal with it later.